### ISSUE TITLE
Firebase storage rules don't reload on save

### ISSUE THREAD

**Issue by DibyodyutiMondal** [firebase/firebase-tools#4366]
### [REQUIRED] Environment info

**firebase-tools:** 10.5.0

**Platform:** Windows 11, node 16

### [REQUIRED] Steps to reproduce

```
rules_version = '2';
service firebase.storage {
  match /b/{bucket}/o {
    match /{file=**} {
      allow read, write: if true;
    }
  }
}
```
The above rule allows all operations

1. Given the above rules, simply run the storage emulator, upload a file (example: `file.txt`). 
2. Then using a new browser tab, go to the emulator storage localhost url for that file (example: `http://localhost:port/v0/b/project-id.appspot.com/o/file.txt`). This should output the json of the object, as expected.
3. Next, change the rules to `allow read, write: if false`. This disables all operations.
The following is logged in the console:
`i  storage: Change detected, updating rules for Cloud Storage...`
4. Reload the browser tab you had previously opened with the object's url. The json response still comes through instead of `403:Forbidden` as it used to.
5. Restart the emulator, then reload the tab for the file
6. New rules now apply

**Comment by AntoninBeaufort** [comment]
I can confirm that I have the same problem on same versions (firebase-tools, Windows and Node but I'm using WSL 2 with Ubuntu 20.04), need to restart emulators to get rules updated.

**Comment by yoavTaieb** [comment]
Same problem on Mac Monterey

**Comment by tohhsinpei** [comment]
Thanks for the report. This should be fixed in #4337 which was included in yesterday's release. Let me know if the problem persists; otherwise closing the issue.
